crates/metal-codegen-llvm/src/safeties.rs

```diff
@@ -33,3 +36,35 @@ impl Drop for MemoryBuffer {
         unsafe { LLVMDisposeMemoryBuffer(self.llvm_buf) };
     }
 }
+
+pub struct LLVMErrorMessage {
+    ptr: *mut c_char,
+}
+
+impl Default for LLVMErrorMessage {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl LLVMErrorMessage {
+    pub fn new() -> Self {
+        Self {
+            ptr: std::ptr::null_mut(),
+        }
+    }
+
+    pub fn llvm(&self) -> *mut i8 {
```

Comment:
should return c_char, also the name "llvm" says little about what the function does. consider adding docs or renaming it